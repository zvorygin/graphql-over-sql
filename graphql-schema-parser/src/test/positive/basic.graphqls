schema {
    query: Query
}

type Query {
    someIntField(arg:Input): Int
    someStringField: String
}

input Input {
    first : Int
    second: Float
}

interface Interface {
    first : Int
    second : Float
}