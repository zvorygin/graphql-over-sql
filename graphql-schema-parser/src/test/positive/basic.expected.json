{
  "schema": {
    "queryType": {
      "collection": false,
      "nonNull": false,
      "typeName": "Query"
    }
  },
  "types": {
    "Query": {
      "fields": [
        {
          "name": "someIntField",
          "type": {
            "collection": false,
            "nonNull": false,
            "typeName": "Int"
          }
        },
        {
          "name": "someStringField",
          "type": {
            "collection": false,
            "nonNull": false,
            "typeName": "String"
          }
        }
      ],
      "name": "Query"
    }
  },
  "interfaces": {
    "Interface": {
      "fields": [
        {
          "name": "first",
          "type": {
            "collection": false,
            "nonNull": false,
            "typeName": "Int"
          }
        },
        {
          "name": "second",
          "type": {
            "collection": false,
            "nonNull": false,
            "typeName": "Float"
          }
        }
      ],
      "name": "Interface"
    }
  },
  "scalars": {
    "Float": {
      "name": "Float"
    },
    "String": {
      "name": "String"
    },
    "Boolean": {
      "name": "Boolean"
    },
    "Int": {
      "name": "Int"
    }
  },
  "inputs": {
    "Input": {
      "fields": [
        {
          "name": "first",
          "type": {
            "collection": false,
            "nonNull": false,
            "typeName": "Int"
          }
        },
        {
          "name": "second",
          "type": {
            "collection": false,
            "nonNull": false,
            "typeName": "Float"
          }
        }
      ],
      "name": "Input"
    }
  }
}